Make.defs

```diff
@@ -100,6 +100,14 @@ define REGISTER
 endef
 endif
 
+# Download
+
+define DOWNLOAD
```

Comment:
should we move to nuttx/tools/Config.mk? so all place incuding nuttx can use this macro.

Reply:
That is up to you, but I suspected that it might not be desirable to download in NuttX itself. At the same time, providing it to anyone through NuttX might not do any harm.

For me, the inclusion in applications is enough. I am always using NuttX with applications and adding my software as external applications. Thus I can use this in my external applications.

Reply:
> That is up to you, but I suspected that it might not be desirable to download in NuttX itself. At the same time, providing it to anyone through NuttX might not do any harm.
> 
> For me, the inclusion in applications is enough. I am always using NuttX with applications and adding my software as external applications. Thus I can use this in my external applications.

 nuttx repo has many external project downloaded by curl too, we can't reuse the same macro in nuttx if DOWNLOAD is defined in Config.mk.

Reply:
I haven't check. If so then let's move it to NuttX then.

Reply:
The updated version now uses `DOWNLOAD` from Nuttix, submitted over here: https://github.com/apache/nuttx/pull/8759.

I had to tweak it to allow different output, and I removed the prefix to the reported output. I think that the correct approach is to use the output (the third argument of the `DOWNLOAD` variable). The nothing saying downloads (those have versions as file names) should be replaced by a custom output name. I just skip doing that as part of this PR and we can decide later if we want that.